### ISSUE TITLE
Sitemaps are only generated for content in the default culture

### ISSUE THREAD

**Issue by jagbarcelo** [OrchardCMS/OrchardCore#7440]
According to Google documentation at https://support.google.com/webmasters/answer/189077?hl=en

![image](https://user-images.githubusercontent.com/6704109/97635698-118ee400-1a38-11eb-9196-5e82ca49d4b8.png)

This means that a sitemap should include all combinations from **any** culture (`<loc>`) to **any other** culture (`<xhtml:link...>`).

In our case, it is only generated from en-GB (our default culture) to our alternative culture es-ES. In other words, the sitemap does not include any `<loc>` tags for the Spanish version (and it should). It only has `<loc>` tags for the en-GB culture.

It is not enough to list the English pages (using `<loc>`) and their matching counterparts (using `<xhtml:link...>`). All pages, no matter their culture, should be listed.

A side effect of this issue is that a sitemap.xml file posted to Bing Webmaster Tools (that does not support `<xhtml:link...>` tags), does not crawl our Spanish pages (only the English ones, since there are no `<loc>` entries for the Spanish version).

**Comment by jagbarcelo** [comment]
I do not know how to fix it, but I have pin-pointed the cause:

![image](https://user-images.githubusercontent.com/6704109/97710163-ad156880-1abb-11eb-93a2-bfbe33d15658.png)

The contentItem passed as parameter cannot be casted to a `SitemapPart` and because of this `contentItem.As<SitemapPart>()` returns null. In this case contentItem is a Spanish page, being the site default culture another one.

Because of this ValidateContentItem returns false, and later on, isValid == false (in ContentTypesSitemapSourceBuilder.cs) and thus, this contentItem is ignored.

![image](https://user-images.githubusercontent.com/6704109/97710606-3cbb1700-1abc-11eb-85a8-f968fb6753bf.png)

By the way... this `foreach` loop will discard the results of every `validationProvider` except for the last one. I do not know either if this is intended or might be another mistake.

**Comment by deanmarcussen** [comment]
Looks like I got a condition wrong there somewhere.

I think off the top of my head `isValid` should be true by default though.

Let me try and remember why / how it was written like that, because it suggests that nothing would turn up in the sitemap, where you are only missing one of the cultures?

i.e. you have En, and it's not showing Es?

**Comment by jagbarcelo** [comment]
Yes, we have two languages: `en-GB` (default culture) and also `es-ES`. Only `en-GB` pages are being included in the sitemap in `<loc>` tags.

**Comment by jagbarcelo** [comment]
In your last message, your second output (the one that should be producing) was not correctly formatted for the GitHub editor and it was a little scrambled.

Anyway, in a site with two languages and a total of two pages (1 language each page) there should be two `<loc>` tags and each of them should list all its alternates (including itself) inside `<xhtml:link>` tags.

Google's example (see the URL at the top of this thread) has a total of 3 pages, with 3 languages (1 language for each page) and it has 3 `<loc>` tags.

![image](https://user-images.githubusercontent.com/6704109/97743975-bc5ddb80-1ae6-11eb-9fc5-0a8bca23572c.png)

Summing up... the combination of (all pages) x (all alternates).

**Comment by sebastienros** [comment]
I fixed the code block, and it also looks like there is an agreement now.
